lib/Conch/Controller/Rack.pm

```diff
@@ -344,6 +344,34 @@ sub delete_assignment ($c) {
     return $c->status(204);
 }
 
+=head2 set_phase
+
+Updates the phase of this rack, and optionally all devices located in this rack.
+
+Use the C<rack_only> query parameter to specify whether to only update the rack's phase, or all
+located devices' phases as well.
+
+=cut
+
+sub set_phase ($c) {
+    my $input = $c->validate_input('RackPhase');
+    return if not $input;
+
+    $c->stash('rack_rs')->update({ phase => $input->{phase}, updated => \'now()' });
+    $c->log->debug('set the phase for rack '.$c->stash('rack_id').' to '.$input->{phase});
+
+    if (not $c->req->query_params->param('rack_only') // 0) {
```

Comment:
The `clear_tokens` param also allows `no` and `false` as false values. Should we be consistent between these two endpoints?

Reply:
they're already inconsistent: one is a bool and the other isn't.